stix2/observables.py

```diff
@@ -184,11 +281,69 @@ class Mutex(_Observable):
     }
 
 
+class HTTPRequestExt(_STIXBase):
+    _properties = {
+        'request_method': StringProperty(),
+        'request_value': StringProperty(),
+        'request_version': StringProperty(),
+        'request_header': DictionaryProperty(),
+        'message_body_length': IntegerProperty(),
+        'message_body_data_ref': ObjectReferenceProperty(),
```

Comment:
_check_property() in _Observable should check object references inside extensions. We'll need to add a test for this.